src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6073,4 +6073,25 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="IDS_FeatureModuleInitializers" xml:space="preserve">
     <value>module initializers</value>
   </data>
+  <data name="ERR_ModuleInitializerMethodMustBeAccessibleOutsideTopLevelType" xml:space="preserve">
+    <value>Module initializer method '{0}' must be accessible outside top-level type '{1}'</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustBeStatic" xml:space="preserve">
+    <value>Module initializer method '{0}' must be static</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustNotBeContainedInGenericType" xml:space="preserve">
+    <value>Module initializer method '{0}' must not be contained in a generic type</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustNotBeGeneric" xml:space="preserve">
+    <value>Module initializer method '{0}' must not have generic parameters</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustNotHaveParameters" xml:space="preserve">
+    <value>Module initializer method '{0}' must not have parameters</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustReturnVoid" xml:space="preserve">
+    <value>Module initializer method '{0}' must return void</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustBeOrdinary" xml:space="preserve">
+    <value>A module initializer must be an ordinary method</value>
```

Comment:
>an ordinary method [](start = 40, length = 18)

I do not believe language specification has a concept of "an ordinary method". What exactly are we trying to disallow here? Perhaps we should just name them: constructor, ...

Reply:
@RikkiGibson said the same thing. I had a question about this and would appreciate your input: https://github.com/dotnet/roslyn/pull/43301#discussion_r417364225

Reply:
> I had a question about this and would appreciate your input.

I think there is a suggestion to enumerate unsupported possible targets in my original comment 

---
In reply to: [418180150](https://github.com/dotnet/roslyn/pull/43301#discussion_r418180150) [](ancestors = 418180150)

Reply:
I'm a little confused. Is "Perhaps we should just name them: constructor, ..." the suggestion you are referring to? My linked question is how to keep the list up to date as the language evolves.

Reply:
> My linked question is how to keep the list up to date as the language evolves.

Usually, when something new comes along, we should test that it is supported/unsupported with other features. During this process, we discover that the message should be adjusted. There is always an option to leave a comment somewhere, like in MethodKind enum, but that is not common.   

---
In reply to: [418195437](https://github.com/dotnet/roslyn/pull/43301#discussion_r418195437) [](ancestors = 418195437)

Reply:
Marking resolved since your suggestion in https://github.com/dotnet/roslyn/pull/43301#discussion_r418233735 was to go with the phrase "an ordinary member method" due to the length of the list.